### PR TITLE
BUG-25061 fix printing indices with NaNs

### LINKED ISSUE #25061: Formatting of CategoricalIndex misaligned if NaNs are present
#### Code Sample, a copy-pastable example if possible

```python
>>> import pandas as pd
>>> pd.DataFrame([[1],[2]], index=pd.CategoricalIndex([1, pd.np.nan]))
      0
 1.0  1
NaN   2
```
#### Problem description

`CategoricalIndex` of a `DataFrame` is misaligned when printing if it contains `NaN` values.

#### Expected Output

```python
>>> import pandas as pd
>>> pd.DataFrame([[1],[2]], index=pd.CategoricalIndex([1, pd.np.nan]))
      0
1.0   1
NaN   2
```

#### Output of ``pd.show_versions()``

<details>

No module named 'dask'

INSTALLED VERSIONS
------------------
commit: da5f5eb6c11140a2bb9f84049d30ba7d89bdbfd0
python: 3.6.6.final.0
python-bits: 64
OS: Darwin
OS-release: 18.0.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_CA.UTF-8
LOCALE: en_CA.UTF-8

pandas: 0.24.0.dev0+1684.gda5f5eb6c
pytest: 4.0.0
pip: 10.0.1
setuptools: 40.6.2
Cython: 0.29
numpy: 1.14.2
scipy: 1.1.0
pyarrow: 0.11.1
xarray: 0.11.0
IPython: 7.1.1
sphinx: 1.8.2
patsy: 0.5.0
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 3.0.1
openpyxl: 2.5.9
xlrd: 1.1.0
xlwt: 1.2.0
xlsxwriter: 1.1.2
lxml.etree: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.14
pymysql: 0.9.2
psycopg2: None
jinja2: 2.10
s3fs: 0.1.6
fastparquet: 0.2.1
pandas_gbq: None
pandas_datareader: None
gcsfs: 0.1.2

</details>

### PR DESCRIPTION
**Pull request by JustinZhengBC**
- [X] closes #25061- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

The `FloatArrayFormatter` leaves a space in front of the floats it formats (expected behaviour as demonstrated by [this test](https://github.com/pandas-dev/pandas/blob/638ddebaa7da1e569836a5b89593b120dbbf491c/pandas/tests/io/formats/test_format.py#L2395)). To avoid misaligned values when printing indices, `NaN` values must be treated similarly.

### CODE REVIEW

File: pandas/tests/series/test_repr.py [hunk 1]
```diff
@@ -198,6 +198,13 @@ def test_latex_repr(self):
 
         assert s._repr_latex_() is None
 
+    def test_categorical_index_repr_in_frame_with_nan(self):
+        i = Index([1, np.nan])
```
Comment (gfyoung) [inline pandas/tests/series/test_repr.py]: Reference issue above this line as a comment.

Comment (gfyoung) [reply pandas/tests/series/test_repr.py]: I meant above the line `i = Index(...)` (hence why I put the comment where I did).  I corrected it myself.

**Review by gfyoung — APPROVED** [review]
cc @jreback

File: pandas/io/formats/format.py [hunk 2]
```diff
@@ -1060,6 +1060,11 @@ def get_result_as_array(self):
         def format_values_with(float_format):
             formatter = self._value_formatter(float_format, threshold)
```
Comment (jreback) [inline pandas/io/formats/format.py]: can you do
``na_rep = self.na_rep`` here, then update (and use that value); so the value attached to self is not changed.

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @JustinZhengBC

**Pull request merged**